src/output.rs

```diff
@@ -6,6 +6,13 @@ use std::string::FromUtf8Error;
 /// Handlebars uses this trait to define rendered output.
 pub trait Output {
     fn write(&mut self, seg: &str) -> Result<(), IOError>;
+
+    /// Designed to be used with `write!` macro.
+    /// for backward compatibility and to avoid breakage the default implementation
+    /// uses `format!` this may be not what you want.
+    fn write_fmt(&mut self, args: std::fmt::Arguments<'_>) -> Result<(), IOError> {
+        self.write(&std::fmt::format(args))
+    }
```

Comment:
The default `impl` shall be here correct me if I'am wrong.

Reply:
By calling `args.as_str()`, it could avoid unnecessary allocation if the content of format arguments has nothing to format.